cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/common/AbstractServiceRetryableMacroEvaluator.java

```diff
@@ -90,15 +90,14 @@ public Map<String, String> evaluateMap(String macroFunction, String... args) thr
           delay = (long) (delay * (minMultiplier + Math.random() * (maxMultiplier - minMultiplier + 1)));
           delay = Math.min(delay, RETRY_MAX_DELAY_MILLIS);
         } catch (IOException e) {
-          throw new RuntimeException("Failed to evaluate the macro function '" + functionName
-                                       + "' with args " + Arrays.asList(args), e);
+          throw new RuntimeException(e);
```

Comment:
Can this happen only if the macro does not exist? Then can we throw `InvalidMacroException` here?

Reply:
Fixed